docs/_docs/04_reference/config-blocks-and-attributes.md

```diff
@@ -401,6 +413,41 @@ dependencies {
 ```
 
 
+### generate
+
+The `generate` block can be used to arbitrarily generate a file in the terragrunt working directory (where `terraform`
+is called). This can be used to generate common terraform configurations that are shared across multiple terraform
+modules. For example, you can use `generate` to generate the provider blocks in a consistent fashion by defining a
+`generate` block in the parent terragrunt config.
+
+The `generate` block supports the following arguments:
+
+- `name` (label): You can define multiple `generate` blocks in a single terragrunt config. As such, each block needs a
+  name to differentiate between the other blocks.
+- `path` (attribute): The path where the generated file should be written. If a relative path, it'll be relative to the
+  Terragrunt working dir (where the terraform code lives).
+- `if_exists` (attribute): What to do if a file already exists at `path`. Valid values are: `overwrite` (overwrite the
+  existing file), `skip` (skip code generation and leave the existing file as-is), `error` (exit with an error).
+- `contents` (attribute): The contents of the generated file.
+
+Example:
+
+```hcl
+# When using this terragrunt config, terragrunt will generate the file "provider.tf" with the aws provider block before
+# calling to terraform. Note that this will overwrite the `provider.tf` file if it already exists.
+generate "provider" {
+  path      = "provider.tf"
+  if_exists = "skip"
```

Comment:
Comment above says we will overwrite, but code example sets `skip`

Reply:
Good catch! Fixed: 667de58